concordium-std/src/impls.rs

```diff
@@ -911,29 +963,26 @@ where
     fn next(&mut self) -> Option<Self::Item> {
         self.state_iter.next().and_then(|entry| {
             let key = entry.get_key();
-            let mut key_cursor = Cursor {
-                data:   key,
-                offset: 8, // Items in a set always start with the set prefix which is 8 bytes.
-            };
-            let res = T::deserial_state_ctx(&self.state_ll, &mut key_cursor).expect_report(
-                "Deserial failed. State has been incorrectly altered using the low-level API.",
-            );
-            Some(res)
+            if let Ok(key) = key {
+                let mut key_cursor = Cursor {
+                    data:   key,
+                    offset: 8, // Items in a set always start with the set prefix which is 8 bytes.
+                };
+                let res = T::deserial_state_ctx(&self.state_ll, &mut key_cursor).expect_report(
+                    "Deserial failed. State has been incorrectly altered using the low-level API.",
+                );
+                Some(res)
+            } else {
+                None
+            }
         })
     }
 }
 
 /// # Trait implementations for Parameter
 impl Read for ExternParameter {
     fn read(&mut self, buf: &mut [u8]) -> ParseResult<usize> {
-        let len: u32 = {
-            match buf.len().try_into() {
-                Ok(v) => v,
-                _ => return Err(ParseError::default()),
-            }
-        };
-        let num_read =
-            unsafe { get_parameter_section(0, buf.as_mut_ptr(), len, self.current_position) };
+        let num_read = self.get_parameter_section(0, buf, self.current_position).unwrap_abort();
```

Comment:
Please don't make unrelated changes such as this. It makes it harder and more time-consuming to review. Aim for minimal changes that _only_ relate to the specific PR 😊

Reply:
I completely agree with this and I am strongly against such changes.

There is already so much subtle stuff going on that we do not need more changes.